includes/class-sensei-course.php

```diff
@@ -3313,6 +3316,44 @@ public static function prerequisite_complete_message() {
 		}
 	}
 
+	/**
+	 * Log an event when a course is initially published, but not on
+	 * subsequently publishes (i.e. if the course is unpublished and
+	 * re-published, do not log another event).
+	 *
+	 * @since 2.1.0
+	 * @access private
+	 *
+	 * @param string  $new_status The new post status.
+	 * @param string  $old_status The old post status.
+	 * @param WP_Post $post       The course.
+	 */
+	public function log_initial_publish_event( $new_status, $old_status, $post ) {
+		if ( 'course' !== $post->post_type || $new_status === $old_status ) {
+			return;
+		}
+
+		$already_published = get_post_meta( $post->ID, 'course_already_published' );
+		$publishing        = 'publish' === $new_status;
+		$unpublishing      = 'publish' === $old_status;
+
+		// If we haven't already published this course, log an event.
+		if ( $publishing && ! $already_published ) {
+			$product_id       = get_post_meta( $post->ID, '_course_woocommerce_product', true );
+			$event_properties = [
+				'module_count' => count( wp_get_post_terms( $post->ID, 'module' ) ),
+				'lesson_count' => $this->course_lesson_count( $post->ID ),
+				'product_id'   => $product_id ? $product_id : -1,
+			];
+			sensei_log_event( 'course_publish', $event_properties );
+		}
+
+		// If we are publishing or unpublishing, mark as already published.
+		if ( ( $publishing || $unpublishing ) && ! $already_published ) {
+			add_post_meta( $post->ID, 'course_already_published', true, true );
```

Comment:
Just curious if this post meta will be deleted when the plugin is uninstalled?

Reply:
Good question!

Since this is post meta being added to a course, it will be deleted on uninstall (if the user has selected that option). All post meta is deleted when its post is deleted, and all courses are deleted on uninstall.

The only caveat is that the courses may be moved to trash first, and then permanently deleted after the configured amount of time (default 30 days). In which case the meta will not be deleted until the course is permanently deleted.